src/main/java/seedu/address/logic/parser/NumberParser.java

```diff
@@ -37,6 +38,10 @@ public Command parse() throws ParseException {
             EditCommand newCommand = (EditCommand) lastCommand;
             newCommand.setIndex(index);
             return newCommand;
+        } else if (lastCommand instanceof DeleteCommand) {
+            DeleteCommand newCommand = (DeleteCommand) lastCommand;
+            newCommand.setIndex(index);
+            return newCommand;
```

Comment:
If `lastCommand` is already an instance of `DeleteCommand`, I do not think there is a need for line 42, it's possible to just `return lastCommand.setIndex(index)`. If this is true, then the entire `parse` function, can be simplified to `return lastCommand.setIndex(index)`

Reply:
right now last command is of type Command, but setIndex is a DeleteCommand method, so I have to cast it to a Deletecommand before i am able to set index

Reply:
got it!